### PR TITLE
Update to LLVM 3.8

### PR DESCRIPTION
**Pull request by lefta**
I don't know if it was planned, but I needed it, so I did it ^^

The prebuilt LLVM packages url and the LLVM svn tag have been updated. I also updated the Ubuntu package url to 16.04 LTS (was 14.04).

A quick test on a project full of errors did not raise any obvious bug, but it will need more testing before being merged.

Let me know if I forgot anything.

EDIT: Your CI Glibc seems too old to pass the build, that's odd... Is Debian not up to date enough or do you need to update?

### CODE REVIEW

File: oclint-scripts/oclintscripts/path.py [hunk 1]
```diff
@@ -64,8 +64,8 @@ class url:
     json_compilation_database = 'https://github.com/oclint/oclint-json-compilation-database.git'
     xcodebuild = 'https://github.com/oclint/oclint-xcodebuild.git'
 
-    clang_prebuilt_binary_for_darwin = 'http://llvm.org/releases/3.7.0/clang+llvm-3.7.0-x86_64-apple-darwin.tar.xz'
-    clang_prebuilt_binary_for_ubuntu_lts = 'http://llvm.org/releases/3.7.0/clang+llvm-3.7.0-x86_64-linux-gnu-ubuntu-14.04.tar.xz'
+    clang_prebuilt_binary_for_darwin = 'http://llvm.org/releases/3.8.0/clang+llvm-3.8.0-x86_64-apple-darwin.tar.xz'
+    clang_prebuilt_binary_for_ubuntu_lts = 'http://llvm.org/releases/3.8.0/clang+llvm-3.8.0-x86_64-linux-gnu-ubuntu-16.04.tar.xz'
```
Comment (UnrealQuester) [inline oclint-scripts/oclintscripts/path.py]: The ci build for linux runs in a linux 14.04 container, but the clang version is build for 16.04.

**Comment by UnrealQuester** [comment]
Thank you for providing a PR for this. Your help is very much appreciated.
Will try it out on my machine to see if I can find any bugs that arise from a new clang version.

File: oclint-scripts/oclintscripts/path.py (hunk 1 above)
Comment (lefta) [inline oclint-scripts/oclintscripts/path.py]: Ow, did not think about this. Should I reverse to 14.04 or do you plan to upgrade?

Comment (UnrealQuester) [inline oclint-scripts/oclintscripts/path.py]: Unfortunately, travis does not support 16.04 yet. Even 14.04 is still considered a 'beta'. So it looks like we would have to revert to 3.8 for 14.04 for now.

**Comment by ryuichis** [comment]
@lefta Do you mind update this line https://github.com/oclint/oclint/blob/master/.travis.yml#L14 to `osx_image: xcode7.3`, and see if the Mac CI build can pass? thanks.

**Comment by lefta** [comment]
Not better...

**Comment by ryuichis** [comment]
@lefta thanks for the effort. 

By looking at the Travis CI error log, I am wondering if https://github.com/oclint/oclint/blob/master/oclint-core/cmake/OCLintConfig.cmake#L16 contributes to the problem. 

Any suggestions are welcomed. And let me see if I can jump on it over the weekend. Travelled a lot recently, and also need to fight with the Houston flooding. :(

**Comment by lefta** [comment]
@ryuichis you're welcome ;)

I'm not a Mac OS developper, so I can't help a lot. But from the travis log there is a obviously conflict between downloaded LLVM and system headers. Disabling the line you pointed out may help, but I don't know if there are any other usefull header in this directory. And this would break building with system LLVM for sure.

If this is possible, I would suggest to keep this line and disable the LLVM standard headers include directories. This way both build methods (bundled and system LLVM) would still work. Unfortunately, I can't help digging this further nor test it.

**Comment by ryuichis** [comment]
@lefta sorry for getting back to you such late. If you are still with me on this P.R., could you remove this line https://github.com/oclint/oclint/blob/master/oclint-core/cmake/OCLintConfig.cmake#L16 and push your changes again? Even though this will create another issue, but it, I believe, will make the travis happy. I will check in the other fix after the travis ci becomes green. Thanks.

**Comment by lefta** [comment]
It's OK now ;)

**Pull request merged**